python/src/cloudstorage/cloudstorage_api.py

```diff
@@ -273,6 +275,185 @@ def listbucket(path_prefix, marker=None, prefix=None, max_keys=None,
   return _Bucket(api, bucket, options)
 
 
+# pylint: disable=too-many-locals, too-many-branches, too-many-statements
+def compose(list_of_files, destination_file,
+            content_type=None, retry_params=None, _account_id=None):
+  '''
+    Runs the GCS Compose on the inputed files.
+    Merges between 2 and 32 files into one file. If any of the previous files had the merge
+    run on them it counts towards the 32 limit.
+  Args:
+    list_of_files: list of file name strings or dictionaries with the following format:
+      {'file_name': REQUIRED name of the file to be merged. Do not include the bucket name,
+       'Generation': OPTIONAL Used to specify what version of a file to use,
+       'IfGenerationMatch': OPTIONAL Used to fail requests if versions don't match}
+    destination_file: Path to the desired output file. Must have the bucket in the path.
+    content_type: Used to specify the content-header of the output.
+                  If None will try to guess based off the first file.
+    retry_params: An api_utils.RetryParams for this call to GCS. If None,
+                  the default one is used.
+    _account_id: Internal-use only.
+
+  Raises:
+    TypeError: If the dictionary for the file list is malformed
+    ValueError: If the number of files is outside the range of 2-1024
+    errors.NotFoundError: If any element in the file list is missing the 'file_name' key
+  '''
+  api = storage_api._get_storage_api(retry_params=retry_params,
+                                   account_id=_account_id)
+  '''
+  Needed until cloudstorage_stub.py is updated to accept compose requests
+  TODO: When patched remove the True flow from this if.
+  '''
+  if 'development' in os.environ.get('SERVER_SOFTWARE', '').lower():
+    def _temp_func(bucket, file_list, destination_file, content_type):
+      '''
+        Dev server stub, remove when the dev server accepts compose requests
+
+      '''
+      with open(destination_file, 'w', content_type=content_type) as gcs_merge:
+        for source_file in file_list:
+          try:
+            with open(bucket + source_file['file_name'], 'r') as gcs_source:
+              gcs_merge.write(gcs_source.read())
+          except errors.NotFoundError:
+            logging.warn('File not found %s, skipping', source_file['file_name'])
```

Comment:
Lets create an issue for adding compose to cloudstorage_stub and cleaning up this code when available.
Also, I think you should fail when not found rather than log-and-ignore (to match production behavior)